components/hab/src/cli/hab/sup.rs

```diff
@@ -64,6 +64,12 @@ pub enum HabSup {
         #[structopt(flatten)]
         remote_sup: RemoteSup,
     },
+    /// Restart a Supervisor without restarting its services
+    #[structopt(no_version)]
+    Restart {
```

Comment:
It pains me that we are adding another command to the `hab sup` family of commands (see comment [here](https://github.com/habitat-sh/habitat/blob/ea661c301f55bc930c3edd0b8ba1122536f5dc0a/components/hab/src/main.rs#L156)). I considered adding a `hab sup-manage` subcommand which would be for subcommands handled by the `hab` binary. However, I was not a fan of the name, and separating it from the other commands felt awkward.

Any thoughts on what good UI/UX would be here without adding another `hab` handled command to `hab sup`?

Reply:
I agree that it's awkward internally, but as far as the desired UX is concerned, this _does_ work. That's not to say there's not another UX we could settle on that make the internal code cleaner. That being said, I'm not sure what that would be right now :frowning_face: